gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallableFactory.java

```diff
@@ -60,8 +60,16 @@ private HttpJsonCallableFactory() {}
 
   private static <RequestT, ResponseT> UnaryCallable<RequestT, ResponseT> createDirectUnaryCallable(
       HttpJsonCallSettings<RequestT, ResponseT> httpJsonCallSettings) {
-    return new HttpJsonDirectCallable<RequestT, ResponseT>(
-        httpJsonCallSettings.getMethodDescriptor(), httpJsonCallSettings.getTypeRegistry());
+    UnaryCallable<RequestT, ResponseT> callable =
+        new HttpJsonDirectCallable<>(
+            httpJsonCallSettings.getMethodDescriptor(), httpJsonCallSettings.getTypeRegistry());
+
+    if (httpJsonCallSettings.getParamsExtractor() != null) {
+      callable =
+          new HttpJsonUnaryRequestParamCallable<>(
+              callable, httpJsonCallSettings.getParamsExtractor());
+    }
```

Comment:
I believe this should always be true. Params Extractor is set for (google.api.http) annotation for backwards compatibility or via the (google.api.routing) annotation

Reply:
Ah no. This is wrong. I didn't realize that `Method.hasHttpBindings()` also checks that there is a pathParameter.